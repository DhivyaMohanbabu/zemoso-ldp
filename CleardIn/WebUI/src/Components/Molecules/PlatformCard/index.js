import React, { useRef } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Button, Grid, Typography } from '@material-ui/core';
import PropTypes from 'prop-types';
import { useHistory } from 'react-router-dom';
import excel from '../../../../assets/images/excel.png';

const useStyles = makeStyles((theme) => ({
  root: {
    width: '200px',
  },
  imageBlock: {
    backgroundColor: theme.palette.clearedIn.grey[200],
    height: '148px',
    borderRadius: '4px 4px 0 0',
    position: 'relative',
  },
  image: {
    height: '62px',
  },
  footer: {
    height: '52px',
    padding: '0 11px',
    borderRadius: '0 0 4px 4px',
    boxShadow: '1px 5px 7px rgb(200,200,200,.6)',
  },
  closeIcon: {
    position: 'absolute',
    top: '8px',
    right: '8px',
    cursor: 'pointer',
  },
  button: {
    height: '32px',
    fontSize: '16px',
    textTransform: 'capitalize',
    padding: '0 20px',
  },
  name: {
    position: 'absolute',
    bottom: '8px',
    left: '8px',
    color: 'white',
    display: 'none',
    maxWidth: '189px',
  },
}));

const PlatformCard = ({ image, title, link }) => {
  const classes = useStyles();

  const history = useHistory();

  const textRef = useRef();
  const buttonsRef = useRef();
  const titleRef = useRef();
  const imageRef = useRef();

  const onHover = () => {
    textRef.current.style.display = 'none';
    buttonsRef.current.style.display = 'block';
    titleRef.current.style.display = 'block';
    imageRef.current.style.backgroundColor = 'rgb(100,100,100,.5)';
  };

  const onLeave = () => {
    textRef.current.style.display = 'block';
    buttonsRef.current.style.display = 'none';
    titleRef.current.style.display = 'none';
    imageRef.current.style.backgroundColor = 'rgb(200,200,200,.2)';
  };

  return (
    <Grid
      container
      className={classes.root}
      data-testid="card"
      onMouseOver={onHover}
      onMouseLeave={onLeave}
    >
      <Grid
        item
        sm={12}
        container
        justify="center"
        alignItems="center"
        className={classes.imageBlock}
        ref={imageRef}
      >
        <Grid item>
          <img className={classes.image} src={image} alt={`${title} logo`} />
          <img className={classes.closeIcon} src={excel} alt="close" />
          <div className={classes.name} ref={titleRef}>
            <Typography noWrap variant="h5">{title}</Typography>
          </div>
        </Grid>
      </Grid>
      <Grid
        container
        item
        alignItems="center"
        sm={12}
        className={classes.footer}
      >
        <Grid container ref={textRef}>
          <Grid item zeroMinWidth>
            <Typography noWrap variant="h5">{title}</Typography>
          </Grid>
        </Grid>
        <Grid
          container
          ref={buttonsRef}
          direction="row"
          spacing={1}
          style={{ display: 'none' }}
        >
          <Grid item sm={6} style={{ display: 'inline-block' }}>
            <Button
              variant="contained"
              color="primary"
              className={classes.button}
              onClick={() => history.push(`/${link}`)}
            >
              Add
            </Button>
          </Grid>
          <Grid item sm={6} style={{ display: 'inline-block' }}>
            <Button variant="outlined" className={classes.button}>
              Preview
            </Button>
          </Grid>
        </Grid>
      </Grid>
    </Grid>
  );
};

PlatformCard.propTypes = {
  image: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  link: PropTypes.string,
};

PlatformCard.defaultProps = {
  link: '',
};

export default PlatformCard;
