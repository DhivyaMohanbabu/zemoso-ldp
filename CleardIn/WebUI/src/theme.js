import { grey } from '@material-ui/core/colors';
import { createMuiTheme } from '@material-ui/core/styles';
// import 'typeface-roboto';

const theme = createMuiTheme({
  palette: {
    primary: {
      main: '#1665d8',
      light: '#b0f1f0',
    },
    secondary: {
      main: '#9ea0a5',
      dark: grey,
    },
    error: {
      main: '#ffada5',

    },
    success: {
      main: '#daefa9',
    },
    info: {
      main: grey[900],
      light: '#fdd570',
      dark: grey[600],
    },
    text: {
      primary: '#1c1c21', // '#3e3f42',
      secondary: '#5c5c5c', // #6b6c6f',
    },
    grey: {
      100: '#f8fafb',
      200: '#f2f4f6',
      300: '#e8ecef',
      400: '#98a9bc',
      500: '#778ca2',
    },
    clearedIn: {
      grey: {
        100: '#f6f6f6',
        200: '#eeeeee',
        300: '#cccccc',
        400: '#949494',
        500: '#666666',
      },
      selectedElementBGColor: '#eaf0ff', // rgba(77, 124, 254, 0.12),
      dark: '#5c5c5c',
      bodyBackground: '#fafafa',
      blue: '#1d4cd7',
    },
    relationShip: {
      vendor: '#daefa9',
      customer: '#fdd570',
      partner: '#b0f1f0',
    },
  },
  typography: {
    fontFamily: ['Roboto', 'Rubik', 'sans-serif'].join(','),
    h1: {
      fontSize: '1.875rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    h2: {
      fontSize: '1.625rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    h3: {
      fontSize: '1.25rem',
      fontWeight: 300,
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    h4: {
      fontSize: '1.125rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    h5: {
      fontSize: '1rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 1.5,
      letterSpacing: 'normal',
    },
    h6: {
      fontSize: '1.25rem',
      fontWeight: 500,
      fontStyle: 'normal',
      lineHeight: 1.05,
      letterSpacing: 'normal',
    },
    subtitle1: {
      fontSize: '0.938rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    subtitle2: {
      fontSize: '0.938rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    caption: {
      fontSize: '0.875rem',
      fontWeight: 500,
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    link: {
      fontSize: '0.875rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 1.5,
      letterSpacing: 'normal',
    },
    body1: {
      fontSize: '0.875rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 1.14,
      letterSpacing: 'normal',
    },
    body2: {
      fontFamily: 'Rubik',
      fontSize: '0.875rem',
      fontWeight: 300,
      fontStyle: 'normal',
      lineHeight: 1.5,
      letterSpacing: 'normal',
    },
    button: {
      fontSize: '0.75rem',
      fontWeight: 500,
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
      textTransform: 'none',
    },
  },
  overrides: {
    MuiButton: {
      root: {
        maxWidth: '200px',
        display: 'block',
      },
      label: {
        whiteSpace: 'nowrap',
        overflow: 'hidden',
        textOverflow: 'ellipsis',
      },
    },
  },
});

export default theme;
