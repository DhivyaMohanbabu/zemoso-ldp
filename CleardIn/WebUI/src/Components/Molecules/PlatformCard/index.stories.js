import React from 'react';
import { select, text } from '@storybook/addon-knobs';
import PlatformCard from '.';

export default {
  title: 'molecule/PlatformCard',
  component: PlatformCard,
};

export const Default = () => {
  const platformOptions = {
    Box: {
      image: 'https://cdn.zeplin.io/5fa102170abc4fbe7467eaa5/assets/e18e8f54-178c-406a-91f2-8b1d83f47cc0.svg',
      title: 'Box',
    },
    GoogleSuite: {
      image: 'https://cdn.zeplin.io/5fa102170abc4fbe7467eaa5/assets/d95b656e-2878-43df-9c4c-382f94892585-3x.png',
      title: 'Google Suite',
    },
    Docusign: {
      image: 'https://cdn.zeplin.io/5fa102170abc4fbe7467eaa5/assets/9e69db6f-95ca-4221-b76f-545a010ea986-3x.png',
      title: 'Docusign',
    },
    Figma: {
      image: 'https://cdn.zeplin.io/5fa102170abc4fbe7467eaa5/assets/55c1d662-ded6-4972-98be-454c2a085e49.svg',
      title: 'Figma',
    },
    Okta: {
      image: 'https://cdn.zeplin.io/5fa102170abc4fbe7467eaa5/assets/9c3444bb-6e74-41b1-80b1-e1d812153448.svg',
      title: 'Okta',
    },
    Slack: {
      image: 'https://cdn.zeplin.io/5fa102170abc4fbe7467eaa5/assets/73eb42dd-a071-48cd-af85-ead7e7c5648e.svg',
      title: 'Slack',
    },
  };
  const selectedObj = select('Platform', Object.keys(platformOptions), 'GoogleSuite');
  const title = text('Title', '');
  return (
    <PlatformCard
      title={title !== '' ? title : platformOptions[selectedObj].title}
      image={platformOptions[selectedObj].image}
    />
  );
};
