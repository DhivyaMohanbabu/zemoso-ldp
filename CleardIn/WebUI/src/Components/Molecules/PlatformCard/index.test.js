import React from 'react';
import '@testing-library/jest-dom';
import { fireEvent, render } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/core/styles';
import PlatformCard from '.';
import theme from '../../../theme';

const allProps = {
  image:
    'https://cdn.zeplin.io/5fa102170abc4fbe7467eaa5/assets/9e69db6f-95ca-4221-b76f-545a010ea986-3x.png',
  title: 'Docusign',
};

it('renders PlatformCard snapshot correctly', () => {
  const { container } = render(
    <ThemeProvider theme={theme}>
      <PlatformCard {...allProps} />
    </ThemeProvider>,
  );
  expect(container).toMatchSnapshot();
});

it('renders PlatformCard onMouseEnter and onMouseLeave correctly', () => {
  const { getByTestId, getByText } = render(
    <ThemeProvider theme={theme}>
      <PlatformCard {...allProps} />
    </ThemeProvider>,
  );
  fireEvent.mouseEnter(getByTestId('card'));
  expect(getByText('Add')).toBeVisible();
  expect(getByText('Preview')).toBeVisible();
  fireEvent.mouseOut(getByTestId('card'));
  expect(getByText('Add')).not.toBeVisible();
  expect(getByText('Preview')).not.toBeVisible();
});
