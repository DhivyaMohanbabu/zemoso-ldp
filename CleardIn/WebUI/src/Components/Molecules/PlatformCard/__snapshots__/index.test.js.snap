// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PlatformCard snapshot correctly 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-root-1 MuiGrid-container"
    data-testid="card"
  >
    <div
      class="MuiGrid-root makeStyles-imageBlock-2 MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-center MuiGrid-grid-sm-12"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <img
          alt="Docusign logo"
          class="makeStyles-image-3"
          src="https://cdn.zeplin.io/5fa102170abc4fbe7467eaa5/assets/9e69db6f-95ca-4221-b76f-545a010ea986-3x.png"
        />
        <img
          alt="close"
          class="makeStyles-closeIcon-5"
          src="test-file-stub"
        />
        <div
          class="makeStyles-name-7"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 MuiTypography-noWrap"
          >
            Docusign
          </h5>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root makeStyles-footer-4 MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-grid-sm-12"
    >
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-zeroMinWidth"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 MuiTypography-noWrap"
          >
            Docusign
          </h5>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
        style="display: none;"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-6"
          style="display: inline-block;"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-6 MuiButton-containedPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Add
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-6"
          style="display: inline-block;"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-6"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Preview
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
